### PR TITLE
refactor(server): make httpApi module less dependent

### PR DESCRIPTION
**Pull request by rockjam**

### CODE REVIEW

File: actor-server/actor-http-api/src/main/scala/im/actor/server/api/http/HttpApiFrontend.scala [hunk 1]
```diff
@@ -26,7 +23,16 @@ final class HttpApi(_system: ActorSystem) extends Extension {
   implicit val ec = system.dispatcher
   implicit val mat = ActorMaterializer()
 
-  val hooks = new HttpApiHookControl
+  private val hooks = new HttpApiHookControl
+
+  def runHooks(): Future[Seq[Route]] = hooks.routesHook.runAll()
+
+  // should we pass implicit actor system here, or the one we have in implicit scope here will work???
+  def registerHook(name: String)(f: ActorSystem ⇒ Future[Route]): Unit = {
```
Comment (rockjam) [inline actor-server/actor-http-api/src/main/scala/im/actor/server/api/http/HttpApiFrontend.scala]: We always use it in following way: `HttpApi(system).registerHook()`, and extension is always initialized with system, so we can use same system in registerHook, right?

File: actor-server/actor-bots/src/main/scala/im/actor/server/bot/http/BotsHttpHandler.scala [hunk 2]
```diff
@@ -1,100 +1,109 @@
...
 
   import JsonFormatters._
   import im.actor.bots.BotMessages.{ BotRequest, BotResponse, BotUpdate }
   import system._
 
-  private val botExt = BotExtension(system)
+  implicit val materializer: Materializer = ActorMaterializer()
+
+  //todo: dangerous dependency. move it somehow
+  private lazy val botExt = BotExtension(system)
```
Comment (rockjam) [inline actor-server/actor-bots/src/main/scala/im/actor/server/bot/http/BotsHttpHandler.scala]: Couldn't find better way. The danger is: we use `BotExtension` in this httpHandler, and this httpHandler is registered on `BotExtension` initialization

File: actor-server/actor-http-api/src/main/scala/im/actor/server/api/http/HttpApiFrontend.scala (hunk 1 above)
Comment (prettynatty) [inline actor-server/actor-http-api/src/main/scala/im/actor/server/api/http/HttpApiFrontend.scala]: right

**Comment by rockjam** [comment]
Pushed to master directly

**Pull request closed without merging**